Fixes layering issues brought by the FoV PR.

Ghommie opened the pull request:
## About The Pull Request
I had been complaining about the FoV PR having some issues for quite some time. The major one being the unintential changes to the layering of several objects which used to stand above mobs but now don't (most notable example being potted plants). I told Azarak about it and he replied fixing it would require some time, which is a cordial way to say he probably won't fix it because of reasons, so I took the matter in my hands (in a lingering demeanor tbh).
Objects and effects with layers higher than the usual mob ones had to have their planes changed, but halfway in the PR I discovered that some large mobs also have layers that are also higher than a portion of these objects, and that further complicated the process. To solve it I had to add two more planes; one assigned to large mobs and other fov-hideables with layers going above the majority of things, another for effects and objects that are also displayed above large mobs & co.
Aware that this solution would add an extra layer of maintenance needed when playing around with planes, I have also added some non-doc comments to the layers.dm file to guide people through the layers of different planes, which is the most I could do at the moment.
While I was at it, I created another layer to use in place of the below mob layer for certain mobs such as pAIs and the swarmer beacon megafauna (not to be confused with the random event one, which is a structure).

I have locally and successfully tested it with a christmas tree, two human mobs, a gas mask, a xeno queen, a potted plant and a recycler.
That said, there is one issue with the FoV I'm currently aware of but I'm not planning to fix in this PR since I don't want to overly cogitate now and that's larva not being able to hide under objects like tables anymore (correct me if I'm wrong): The easiest solution on paper would be making another two planes, but the far greater amount of objects and layers I would have to go through to fix such a small hindrance is such that I'm question if it wouldn't be better for it to be revamped later.

## Why It's Good For The Game
This will fix several common issues with planes and layers, the most mundane ones being potted plants and plastic flaps being displayed under mobs and not over. I have just searched for relative issue reports but I have found none, yet I kind of refuse to believe no one officially reported it, so if you know of a reported issue this PR will fix, please include the number in the comments and I'll include them in this section.

EDIT: This should fix #64078.

## Changelog

:cl:
fix: Fixed several common issues with planes and layers, most noticeably potted plants and plastic flaps being displayed under mobs and not over.
/:cl:

Ghommie committed:
layers and planes

Ghommie committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into Ghommie-tg236

Ghommie committed:
All good now.

Ghommie committed:
disgusting xenos

Ghommie commented:
Oh, I just noticed Azarak is also making a unit test for planes, interesting.

Mothblocks commented:
Looks like someone made a similar unit test, do you want to defer to them instead?

Ghommie commented:
As I said before: I only noticed now, so yes, I'm rolling back the changes and deferring the task to them.
Perhaps I'll fix some of the issues reported by the test pertinent to this PR later to alleviate the burden.

optimumtact commented:
@Ghommie

Ghommie committed:
Merging...

Ghommie commented:
One moment, I have it set to "ready for review" so that the windows build CI is run and I get to know if there's something amiss or absolutely fouled up that should be fixed before it's actually ready (hopefully in a couple hours at most).

Ghommie committed:
Fixed most (not all) incorrect planes and layers detected by the unit test.

Ghommie commented:
Ok, I have fixed most incorrect layers and planes detected by the now-removed unit test. I say most because there are some oddities I didn't care dealing with right now such as a few drops from mobs having the mob layer (which is understandable because they are meant to be displayed above the dead mob, on the other side I'm not making these items FoV-hidden), some HUD objects and plane masters (I wish not to tinker with these too much and maybe break something, they can also be easily blacklisted in the next iteration of the unit test that's being coded by Azarak in #64030).

That said, the PR is ready.

LemonInTheDark commented on code/game/objects/items/weaponry.dm:
```diff
@@ -242,7 +242,7 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 		return INITIALIZE_HINT_QDEL
 
 /obj/item/claymore/highlander/robot/process()
```
do you uh, know why this is on process?

Ghommie replied:
No, but if I had to deal with every bit of awful code I stumbled upon while working on this, things would have gone out of scope pretty fast.
